addons/website_sale/models/product.py

```diff
@@ -451,13 +504,39 @@ class Product(models.Model):
 
     website_url = fields.Char('Website URL', compute='_compute_product_website_url', help='The full URL to access the document through the website.')
 
+    base_unit_count = fields.Float('Base Unit Count', required=True, default=1, help="Display base unit price on your eCommerce pages. Set to 0 to hide it for this product.")
+    base_unit_id = fields.Many2one('website.base.unit', string='Custom Unit of Measure', help="Define a custom unit to display in the price per unit of measure field.")
+    base_unit_price = fields.Monetary("Price Per Unit", currency_field="currency_id", compute="_compute_base_unit_price")
+    base_unit_name = fields.Char("Base unit name", compute='_compute_base_unit_name',
+                                 help='Displays the custom unit for the products if defined or the selected unit of measure otherwise.')
+
+    @api.depends('price', 'lst_price', 'base_unit_count')
+    def _compute_base_unit_price(self):
+        for product in self:
+            product.base_unit_price = product.base_unit_count and (product.price or product.lst_price) / product.base_unit_count or 0
+
+    @api.depends('uom_id', 'base_unit_id', "product_tmpl_id.base_unit_id")
```

Comment:
why not `uom_name`? Doesn't it work since it is a related to `uom_id` ?
And why `product_tmpl_id.base_unit_id` ? (if you keep it use single quote, there is mix between single and double here)